src/solvers/flattening/boolbv.h

```diff
@@ -119,6 +119,10 @@ class boolbvt:public arrayst
   typedef std::unordered_map<const exprt, bvt, irep_hash> bv_cachet;
   bv_cachet bv_cache;
 
+  // record the subexpressions that use the scoped variable,
+  // in order to clear them from the cache after we are done
+  std::vector<bv_cachet::iterator> bv_cache_iterators;
```

Comment:
Iterators into an unordered map aren't stable. Is there a guarantee that we will only ever put iterators in here after the source map isn't modified anymore?

Reply:
Uh, indeed.
This now makes me worried about `boolbvt::convert_bv`. It assumes that `cache_result.first` remains valid during `convert_bitvector`, which I don't believe is the case.

Reply:
Making a few wild guesses: what makes iterators unstable is a possible `rehash` happening. Quite possibly the iterator object itself is ok, as is any reference/pointer contained in there to a value. What will have changed, however, is the relative position of objects, and thus going to the next or previous entry would not work. If this is true, maybe my concerns for this code are unfounded?